drivers/i2c/twihs_sam.c

```diff
@@ -175,6 +187,12 @@ static int twihs_sam_transfer(struct device *dev, struct i2c_msg *msgs,
 		return 0;
 	}
 
+	/* Clear pending interrupts, such as NACK. */
+	(void)twihs->TWIHS_SR;
```

Comment:
Why the `(void)`? Does reading this register clear the interrupts?

Reply:
It is a common practice to put `(void)` in front of what may look like an unnecessary read operation to indicate that we are intentionally ignoring the returned value. In this case reading status register clears all the interrupts. Should I still remove it?

I'll fix other issues.

Reply:
No, it's fine.